pkg/utils/utils.go

```diff
@@ -210,7 +211,16 @@ func CleanCachedNetConf(cRefPath string) error {
 	return nil
 }
 
-// IsValidGUID check if the guild is valid
+// Return GUID string, extracted from hardware address if it is IPoIB, or empty string
+func GetGUIDFromHwAddr(hwAddr net.HardwareAddr) string {
+	const IPoIBAddrLength = 20
+	if len(hwAddr) == IPoIBAddrLength {
+		return hwAddr.String()[36:]
```

Comment:
can you add a comment explaining that the lower 20 bytes represent the GUID in an IPoIB  hardware address

also 
`hw.Addr[12:].String()` might be more readable if you add the comment.